### PR TITLE
Flip Tensor implementation on host and hip

### PR DESCRIPTION
**Pull request by r-abishek**
- Adds rpp support for flip tensor on hip and host.
- Support for U8/F32/F16/I8, NCHW/NHWC/toggle is added.
- Adds corresponding unit tests and performance tests.

### CODE REVIEW

File: src/modules/cpu/kernel/flip.hpp [hunk 1]
```diff
@@ -0,0 +1,1148 @@
...
+        //Compute constant increment, Decrement factors used in source pointer updation
+        Rpp32s srcPtrIncrement = (horizontalFlag)? -vectorIncrement : vectorIncrement;
+        Rpp32u hFlipFactor = (vectorIncrement - 1) * horizontalFlag;
+        Rpp32s srcPtrIncrementPerChannel = (horizontalFlag)? -vectorIncrementPerChannel : vectorIncrementPerChannel;
+        Rpp32u hFlipFactorPerChannel = (vectorIncrementPerChannel - 1) * horizontalFlag;
+        Rpp32s srcPtrIncrementPerRGB = (horizontalFlag) ? -3 : 3;
+        Rpp32s srcPtrIncrementPerPixel = (horizontalFlag) ? -1 : 1;
+        
+        // flip without fused output-layout toggle (NHWC -> NHWC or NCHW -> NCHW and  horizontalflag = 0 and verticalflag = 0)
+        if ((horizontalFlag == 0) && (verticalFlag == 0) && (srcDescPtr->layout == dstDescPtr->layout))
```
Comment (rrawther) [inline src/modules/cpu/kernel/flip.hpp:74]: please change this to
if (!(horizontalFlag|verticalFlag) && (srcDescPtr->layout == dstDescPtr->layout))

File: src/modules/hip/kernel/flip.hpp [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+    {
+        return;
+    }
+
+    d_float24 pix_f24;
+    uint horizontalFlag, verticalFlag, srcIdx;
+    horizontalFlag = horizontalTensor[id_z];
+    verticalFlag = verticalTensor[id_z];
+
+    if(horizontalFlag == 0 && verticalFlag == 0)
```
Comment (rrawther) [inline src/modules/hip/kernel/flip.hpp:28]: These conditional code can be rewritten more efficiently hip kernel. True for all the following code

**Review by rrawther — CHANGES_REQUESTED** [review]
@paveltc to verify

File: src/modules/cpu/kernel/flip.hpp (hunk 1 above)
Comment (r-abishek) [reply src/modules/cpu/kernel/flip.hpp]: done

**Comment by paveltc** [comment]
@rrawther @r-abishek This PR passes all unit tests but it also runs the BatchPD test for u8. The results for it look correct in this one though.

**Comment by r-abishek** [comment]
@rrawther @paveltc Thanks Pavel, the unit tests are expected to run the BatchPD variants too for now. We'll be phasing it out once we reach critical volume on the tensor variants.

**Comment by paveltc** [comment]
@LakshmiKumar23 @rrawther @r-abishek I ran all of the unit tests using the abishekr/rocm5.1.1_ubuntu18.04_opencv4.5.5_rpp_mivisionx:v0.1.0 docker.
Everything looks fine with the new docker.

**Review by rrawther — APPROVED** [review]
@paveltc to verify

**Pull request merged**